sw/host/opentitanlib/src/transport/hyperdebug/spi.rs

```diff
@@ -346,4 +405,28 @@ impl Target for HyperdebugSpiTarget {
         }
         Ok(())
     }
+
+    fn assert_cs<'a>(self: Rc<Self>) -> Result<AssertChipSelect> {
+        {
+            let cs_asserted_count = self.cs_asserted_count.get();
+            if cs_asserted_count == 0 {
+                self.do_assert_cs(true)?;
+            }
+            self.cs_asserted_count.set(cs_asserted_count + 1);
+        }
+        Ok(AssertChipSelect::create(self))
+    }
+}
+
+impl TargetChipDeassert for HyperdebugSpiTarget {
+    fn deassert_cs(&self) {
+        let cs_asserted_count = self.cs_asserted_count.get();
+        if cs_asserted_count - 1 == 0 {
+            // We cannot propagate errors through `Drop::drop()`, so panic on any error.  (Logging
+            // would be another option.)
+            self.do_assert_cs(false)
+                .expect("Error while deasserting CS");
```

Comment:
How will this panic affect the opentitan_session program?  Will this cause program termination?

Depending on the severity of this error (ie: requires manual intervention to fix), panic may be the right action.  However, if destroying and restarting a session works, then `log::error!` might be better.

Ok to leave this as-is for now and figure it out later.